File: iosApp/iosApp/ui/home/RoomListUiState.swift
import Shared

// 공개 방 목록·탐색 (M-11) — Compose RoomListUiState.kt 미러
struct RoomListUiState {
    var isLoading: Bool = true

    var isLoadingMore: Bool = false

    var rooms: [PublicRoom] = []

    var query: String = ""

    var typeFilter: RoomTypeFilter = .all

    var hasNext: Bool = false

    var nextCursor: String? = nil

    var hasError: Bool = false

    var isEmpty: Bool {
        !isLoading && !hasError && rooms.isEmpty
    }
}

File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/ui/home/RoomListUiState.kt
package org.sesacteamproject.passmate.ui.home

import org.sesacteamproject.passmate.payment.domain.model.PublicRoom
import org.sesacteamproject.passmate.payment.domain.model.RoomTypeFilter

// 공개 방 목록·탐색 화면 상태 (M-11)
data class RoomListUiState(
    val isLoading: Boolean = true,
    val isLoadingMore: Boolean = false,
    val rooms: List<PublicRoom> = emptyList(),
    val query: String = "",
    val typeFilter: RoomTypeFilter = RoomTypeFilter.ALL,
    val hasNext: Boolean = false,
    val nextCursor: String? = null,
    val hasError: Boolean = false
) {
    val isEmpty: Boolean
        get() = !isLoading && !hasError && rooms.isEmpty()
}
